sqlvm-provisioning-csp/SQLServerVMDeployment.json

```diff
@@ -0,0 +1,238 @@
+{
+    "$schema": "https://schema.management.azure.com/schemas/2015-01-01/deploymentTemplate.json#",
+    "contentVersion": "1.0.0.0",
+    "parameters": {
+        "vmName": {
+            "type": "string",
+            "defaultValue": "",
+            "metadata": {
+                "description": "SQL IaaS VM machine name"
+            }
+        },
+        "vmSize": {
+            "type": "string",
+            "defaultValue": "",
+            "metadata": {
+                "description": "SQL IaaS VM size. Use Azure VM size name from MSDN"
+            }
+        },
+        "sqlImageOffer": {
+            "type": "string",
+            "defaultValue": "SQL2014-WS2012R2",
+            "allowedValues": [
+                "SQL2014-WS2012R2"
+            ],
+            "metadata": {
+                "description": "SQL Server Gallery Image Offer"
+            }
+        },
+        "sqlImageSku": {
+            "type": "string",
+            "defaultValue": "Standard",
+            "allowedValues": [
+                "Enterprise",
+                "Standard",
+                "Web"
+            ],
+            "metadata": {
+                "description": "SQL Server Gallery Image SKU"
+            }
+        },
+        "sqlImageVersion": {
+            "type": "string",
+            "defaultValue": "latest",
+            "metadata": {
+                "description": "SQL Server Gallery Image Published Version"
+            }
+        },
+        "vmLocation": {
+            "type": "string",
+            "defaultValue": "",
+            "metadata": {
+                "description": "SQL IaaS VM data center location. User Azure location from MDSN"
+            }
+        },
+        "username": {
+            "type": "string",
+            "defaultValue": "",
+            "metadata": {
+                "description": "SQL IaaS VM local administrator username"
+            }
+        },
+        "password": {
+            "type": "securestring",
+            "defaultValue": "",
+            "metadata": {
+                "description": "SQL IaaS VM local administrator password"
+            }
+        },
+        "storageName": {
+            "type": "string",
+            "defaultValue": "",
+            "metadata": {
+                "description": "SQL IaaS VM data and OS disks storage service"
+            }
+        },
+        "storageType": {
+            "type": "string",
+            "defaultValue": "Standard_LRS",
+            "allowedValues": [
+                "Standard_LRS",
+                "Standard_GRS",
+                "Standard_ZRS"
+            ],
+            "metadata": {
+                "description": "SQL IaaS VM data and OS disks storage type"
+            }
+        },
+        "vnetName": {
```

Comment:
This can become a variable as this doesn't need to globally unique and reduces the number of parameters a user has to enter